rerun_py/rerun_sdk/rerun/_rerun2/_validators.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+
+
+# This code follows closely from `image_ext.rs`
+def find_non_empty_dim_indices(shape: list[int]) -> list[int]:
+    """Returns the indices of an appropriate set of non-empty dimensions."""
+    if len(shape) < 2:
+        return list(range(len(shape)))
+
+    indices = list(d[0] for d in filter(lambda d: d[1] != 1, enumerate(shape)))
+
+    # 0 must be valid since shape isn't empty or we would have returned an Err above
+    first_non_empty = next(iter(indices), 0)
+    last_non_empty = next(reversed(indices), first_non_empty)
```

Comment:
what python magic is this - why not use `indices[0]` instead of `next(iter(` ?

Reply:
indices[0] / indices[-1] will raise on empty list. 

This would happen, for example, with shape [1,1,1] since we'll filter them all out.